test/extern/ferguson/int-to-ptr.chpl

```diff
@@ -0,0 +1,16 @@
+use SysCTypes;
+use CPtr;
+
+var a: c_void_ptr = (-1): c_void_ptr;
+var b: c_void_ptr = (-1): int : c_void_ptr;
+var c: c_void_ptr = (-1): uint : c_void_ptr;
+var d: c_void_ptr = (-1): c_intptr : c_void_ptr;
+var e: c_void_ptr = (-1): c_uintptr : c_void_ptr;
```

Comment:
I feel as though this test would be stronger if it did some tests of both 32- and 64-bit ints / uints since things like the where clauses in the section of code this PR touches would presumably evaluate differently with back-end compilers that used 32- vs. 64-bit ints and/or pointers (?).  For example, I could imagine collapsing your five cases down to something like:

```chapel
proc testType(type t) {
  param x: t = -1;
  var y: t = x;
  var a: c_void_ptr = x: c_void_ptr;
  var b: c_void_ptr = (-1): t : c_void_ptr;  // maybe this case is superfluous now or should be something else?
  var c: c_void_ptr = x: c_intptr : c_void_ptr;
  var d: c_void_ptr = x: c_uintptr : c_void_ptr;

  assert(a == b);
  assert(b == c);
  assert(c == d);
}

proc testWidth(param w) {
  testType(int(w));
  testType(uint(w));
}

testWidth(32);
testWidth(64);
testWidth(8);  // why not?
```

Reply:
I added to the test. I need to check to see if it works on a 32-bit system. I'm worried the casting from a 64-bit int to a 32-bit pointer will lead to truncation that will cause it to fail, so I'll look at that.

Reply:
It'd be fine with me to merge and let nightly testing take care of checking that case, given the hassles of 32-bit systems.  I'd convinced myself that with -1 as the value, the truncation would probably "just work" (where it might not for other more interesting values), but may have been thinking sloppily.

Reply:
I've checked it on a 32-bit system and updated the test. Is this PR good to go or do you have other feedback?